src/tools/pkcs15-init.c

```diff
@@ -1782,6 +1788,9 @@ do_generate_key(struct sc_profile *profile, const char *spec)
 	if (r == 0)
 		r = sc_pkcs15init_generate_key(g_p15card, profile, &keygen_args, keybits, NULL);
 	sc_unlock(g_p15card->card);
+	if (algorithm == SC_ALGORITHM_EC)
+		/* allocated in parse_alg_spec() */
+		free(keygen_args.prkey_args.key.u.ec.params.named_curve);
```

Comment:
It looks strange to have this here.

In `parse_alg_spec()`, would it cause any problems if we take (part of) the user supplied spec for `u.ec.params.named_curve` avoiding the `strdup()` in the first place?

Reply:
I replaced the `free` with `sc_pkcs15_free_prkey()`. Removing `strdup()` would probably also mean to remove `const` from `spec` param and I!m not sure about all places where `named_curve` is used then.